pkg/tnf/handlers/podantiaffinity/podantiaffinity.json

```diff
@@ -0,0 +1,29 @@
+{
+  "identifier": {
+    "url": "http://test-network-function.com/tests/podantiaffinity",
+    "version": "v1.0.0"
+  },
+  "description": "This test checks cnf pod antiaffinity rule in cnf deployment.",
+  "testResult": 0,
+  "testTimeout": 10000000000,
+  "reelFirstStep": {
+    "execute": "oc get deployment {{.DEPLOYMENT_NAME}} -n {{.DEPLOYMENT_NAMESPACE}} -o json | jq -r ' if .spec.replicas > 1 then if .spec.template.spec.affinity.podAntiAffinity != null then \"OK\" else \"MISSING\" end else \"OK\" end '\n",
```

Comment:
this is probably not recommended. It would be cleaner to have two handlers

Reply:
@jun-rh  probably something like this ?
```
oc get -n sqlnamespace deployment/sql -o go-template='{{ .spec.replicas }}{{" "}}{{len  .spec.template.spec.affinity.podAntiAffinity }}{{"\n"}}'
```

Reply:
@HamadiSE the go-template syntex is not working when podAntiAffinity  is not present in json.
@jun-rh , not fully get the need to have two handlers when one query and filter can process the the context.

Reply:
@wying3 In general, this kind of code embedded in a command filter is not very reusable or extensible. I had a hard time trying to tell whether the return values make sense from this one line nested if block. Is the last ok returned when replica count is 1? If that's the case, it would be wrong. We want to fail the test if the count is not greater than 1. Think about maintainability, would you continue this approach when we need to add a third or fourth check?

Reply:
BTW, in theory, we need to check not just the presence of the antiaffinity rule, but also the content of the rule. There has to be at least one rule that is against the pod's own label with RequiredDuringScheduling and Node topology. Can this be done in a jq expression in a clean way? Maybe even the json based generic test becomes too limited for that kind of check.